flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/InternalTimersSnapshotReaderWriters.java

```diff
@@ -223,10 +167,10 @@ public static <K, N> InternalTimersSnapshotReader<K, N> getReaderForVersion(
 
         switch (version) {
             case NO_VERSION:
-                return new InternalTimersSnapshotReaderPreVersioned<>(userCodeClassLoader);
-
             case 1:
-                return new InternalTimersSnapshotReaderV1<>(userCodeClassLoader);
+                throw new UnsupportedOperationException(
```

Comment:
`IllegalStateException`?

Reply:
This is the reading code path, so I think `UnsupportedOperationException` is better suited here?

Reply:
I don't have strong opinion here, so we can go with either.

My preference is for `IllegalStateException` because
1. It is consistent with the `default` branch
2. I see `UnsupportedOperationException` rather to be thrown in some stubs/mocks that have limited functionality and I see it more likely to be supported at some point. This case is not a limited functionality, but a wrong state/argument(savepoint) that has been passed.

Still, I want to emphasize I am good with either.

Reply:
I will keep it as it is. The default branch is an illegal state, as if it's triggered that's a bug in the code. Someone forgot to implement something. This branch is just not supported (and will never be), but that's a different message to "encountered illegal state".